addons/l10n_jo_edi/models/account_move.py

```diff
@@ -0,0 +1,258 @@
+
+    @api.depends("country_code", "move_type")
+    def _compute_l10n_jo_edi_is_needed(self):
+        for move in self:
+            move.l10n_jo_edi_is_needed = (
+                move.country_code == "JO"
+                and move.move_type in ("out_invoice", "out_refund")
+            )
+
+    @api.depends("l10n_jo_edi_is_needed", "state", "l10n_jo_edi_xml_attachment_id", "l10n_jo_edi_error", "reversal_move_id", "payment_state")
+    def _compute_jo_edi_state(self):
+        jo_moves = self.filtered(lambda move: move.l10n_jo_edi_is_needed)
+        for move in jo_moves:
+            if move.state == 'posted':
+                move.l10n_jo_edi_state = 'to_send'
+            if move.l10n_jo_edi_xml_attachment_id and not move.l10n_jo_edi_error:
+                move.l10n_jo_edi_state = 'sent'
+            if move.reversal_move_id and move.payment_state != 'partial':
+                move.l10n_jo_edi_state = 'to_cancel'
+            if move.state == 'cancel' or move.payment_state == 'reversed':
+                move.l10n_jo_edi_state = 'cancelled'
+            if move.state == 'draft':
+                move.l10n_jo_edi_state = None
+
+    @api.depends("l10n_jo_edi_xml_attachment_id", "l10n_jo_edi_error")
+    def _compute_show_reset_to_draft_button(self):
+        super()._compute_show_reset_to_draft_button()
+        self.filtered(lambda move: move.l10n_jo_edi_xml_attachment_id and not move.l10n_jo_edi_error).show_reset_to_draft_button = False
+
+    def button_draft(self):
+        # EXTENDS 'account'
+        self.write(
+            {
+                "l10n_jo_edi_error": False,
+                "l10n_jo_edi_xml_attachment_file": False,
+                "l10n_jo_edi_xml_attachment_id": False,
+            }
+        )
+        return super().button_draft()
+
+    def _build_jofotara_headers(self):
+        if not self.company_id.l10n_jo_edi_client_id:
+            raise ValidationError("Client ID is missing")
+        if not self.company_id.l10n_jo_edi_secret_key:
+            raise ValidationError("Secret key is missing")
+        return {
+            'Client-Id': self.company_id.l10n_jo_edi_client_id,
+            'Secret-Key': self.company_id.l10n_jo_edi_secret_key,
+            'Content-Type': 'application/json',
+        }
+
+    def submit_to_jofotara(self):
+        self.ensure_one()
+        headers = self._build_jofotara_headers()
+        params = '{"invoice": "%s"}' % self.l10n_jo_edi_xml_attachment_id.datas.decode('ascii')
+
+        try:
+            response = requests.post(JOFOTARA_URL, data=str(params), headers=headers, timeout=50, verify=False)
+        except (requests.exceptions.Timeout, requests.exceptions.RequestException, requests.exceptions.HTTPError) as e:
+            raise UserError("Invalid request: %s" % e)
+
+        response_text = response.content.decode('utf-8')
+        if not response.ok:
+            raise UserError("Request failed: %s" % response_text)
```

Comment:
You must not raise anything to the send and print wizard. Make all those methods returning a dict that could contain an `error` instead